### PR TITLE
Fix pause time of random mobile conversation

### PR DESCRIPTION
**Pull request by pklloveyou**
## Tests Performed:
-```.go -1521 7925 -20 530```
-Talk to the businessman

## How to Test the Changes:
-```.go -1521 7925 -20 530```
Businessmen will not continue to walk

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
hi @pklloveyou can you please add more information? also how to test this?

**Comment by pklloveyou** [comment]
When I finish it

**Comment by pklloveyou** [comment]
@FrancescoBorzi Is this ok?

**Comment by FrancescoBorzi** [comment]
@pklloveyou yes it's ok, thanks for your contribution. Why closing it?

**Comment by FrancescoBorzi** [comment]
@azerothcore/testers please check

File: src/server/game/Movement/MovementGenerators/RandomMovementGenerator.cpp [hunk 1]
```diff
@@ -272,6 +272,17 @@ bool RandomMovementGenerator<Creature>::DoUpdate(Creature* creature, const uint3
         }
     }
 
+    if (!_nextMoveTime.Passed())
```
Comment (Viste) [inline src/server/game/Movement/MovementGenerators/RandomMovementGenerator.cpp:275]:
```
if (!_nextMoveTime.Passed()) && CanMove(diff))
{
    return true;
}```

**Comment by Yehonal** [comment]
It should also fix the same issue when you aggro a creature and this creature continue to further walk for some seconds. Could you test it?

**Comment by pklloveyou** [comment]
I'll try it now

**Comment by pklloveyou** [comment]
> It should also fix the same issue when you aggro a creature and this creature continue to further walk for some seconds. Could you test it?

No,

**Comment by pklloveyou** [comment]
It doesn't fix the mistake you said

File: src/server/game/Movement/MovementGenerators/RandomMovementGenerator.cpp [hunk 2]
```diff
@@ -280,9 +280,7 @@ bool RandomMovementGenerator<Creature>::DoUpdate(Creature* creature, const uint3
     else
     {
         if (creature->IsStopped())
-        {
             _nextMoveTime.Reset(sWorld->getIntConfig(CONFIG_WAYPOINT_MOVEMENT_STOP_TIME_FOR_PLAYER) * IN_MILLISECONDS);
-        }
```
Comment (FrancescoBorzi) [inline src/server/game/Movement/MovementGenerators/RandomMovementGenerator.cpp]: why?

Comment (pklloveyou) [reply src/server/game/Movement/MovementGenerators/RandomMovementGenerator.cpp]: I think you should use this option when you stop moving, right? Terrible translation.

**Pull request closed without merging**